### PR TITLE
validation and tests in `CreateNewImageToolTask`

### PR DESCRIPTION
**Pull request by vlada-shubina**
Implemented validation and added tests for generating parameters for `containerize`

TODO:

- [x] validation and checks for `Labels`, `ImageTags`, `ExposedPorts`, `ContainerEnvironmentVariables`
- [x] https://github.com/dotnet/sdk-container-builds/issues/172

### CODE REVIEW

File: Microsoft.NET.Build.Containers/Tasks/CreateNewImageToolTask.cs [hunk 1]
```diff
@@ -66,35 +66,139 @@ protected override ProcessStartInfo GetProcessStartInfo(string pathToTool, strin
...
+            builder.AppendSwitchIfNotNull("--outputregistry ", OutputRegistry);
+        }
+        if (!string.IsNullOrWhiteSpace(LocalContainerDaemon))
+        {
+            builder.AppendSwitchIfNotNull("--localcontainerdaemon ", LocalContainerDaemon);
+        }
+
+        if (EntrypointArgs.Any(e => string.IsNullOrWhiteSpace(e.ItemSpec)))
+        {
+            //Log.LogWarning($"Item '{nameof(EntrypointArgs)}' contains empty items which will be ignored.");
```
Comment (vlada-shubina) [inline Microsoft.NET.Build.Containers/Tasks/CreateNewImageToolTask.cs:133]: not able to send the warnings from here, need investigation.

**Comment by vlada-shubina** [comment]
Refactoring and tests for `CreateNewImageToolTask` are completed, however to fix the flow of `Labels`, `ExposedPorts`, `ContainerEnvironmentVariables` changes to `containerize` parsing are needed too.

**Review by baronfel — APPROVED** [review]
This is great! Amazing work bringing this up to speed.

File: Directory.Packages.props [hunk 2]
```diff
@@ -3,6 +3,7 @@
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
   </PropertyGroup>
   <ItemGroup>
+    <PackageVersion Include="FakeItEasy" Version="7.3.1" />
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply Directory.Packages.props:6]: This is test only dependency and doesn't need to be in source build. dotnet/sdk is already using this lib.

[the remark this answers is not in the document]
Comment (baronfel) [reply Directory.Packages.props:6]: Thank you for the clarification - in that case this is good to merge.

**Pull request merged**